src/components/custom-field-input-text/custom-field-input-text.jsx

```diff
@@ -0,0 +1,64 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import cautionSvg from '../../svgs/icon-caution-fill.svg';
+import Icon from '../icon/icon.jsx';
+import styles from './custom-field-input-text.css';
+
+function getRootClassName(className, error, disabled) {
+  if (disabled) {
+    return `${className} ${styles.disabled}`;
+  }
+
+  if (error) {
+    return `${className} ${styles.error}`;
+  }
+
+  return className;
+}
+
+export default function CustomFieldInputText(props) {
+  return (
+    <div className={getRootClassName(props.className, props.error, props.disabled)}>
+      <div className={styles['heading-container']}>
+        <label className={styles.label} htmlFor={props.id}>Input Descriptor</label>
+        {props.required && <span className={styles.optional}>(Required)</span>}
+      </div>
+      <div className={styles['input-container']}>
+        <input className={styles.input} disabled={props.disabled} type="text" id={props.id} name={props.name} placeholder={props.placeholder} value={props.value} onClick={props.onClick} />
+        {props.error &&
+          <div className={styles['input-icon-container']}>
+            <Icon className={styles['input-icon']} currentColor="caution" name={cautionSvg.id} size="medium" />
+          </div>
+        }
+      </div>
+      <span className={styles.help}>{props.helpText}</span>
+    </div>
+  );
+}
+
+CustomFieldInputText.propTypes = {
+  className: PropTypes.string,
+  disabled: PropTypes.bool,
+  error: PropTypes.bool,
```

Comment:
I just found out a cool thing (but I am not sure if it works for us):

https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/HTML5/Constraint_validation

We can set the validity of an input via JavaScript. This means we could set the input to be invalid and allow for styling to be based on its pseudo state: `.input:invalid { ... }` type of thing! The method is `setCustomValidity`.